### PR TITLE
Webpack 4

### PR DESCRIPTION
**Pull request by rupertbates**
## Why are you doing this?
To upgrade Webpack to version 4

[**Trello Card**](https://trello.com/c/g8oUw2RT/1394-upgrade-to-webpack-4)

## Changes

* Upgrade webpack and associated dependencies
* Swap ExtractTextPlugin for MiniCssExtractPlugin
* Break config out into Dev and Prod
* Fix the asset path for the dev-server 

### Asset size comparison between Webpack 3 & 4
screen shot 2018-03-26 at 12 20 52

### Tested in 
IE 11
Safari 8
Latest Chrome & Firefox on Mac
Chrome on Android 7
Safari on iPhone 7

### CODE REVIEW

File: app/wiring/Assets.scala [hunk 1]
```diff
@@ -4,5 +4,5 @@ import assets.AssetsResolver
 import play.api.BuiltInComponentsFromContext
 
 trait Assets { self: BuiltInComponentsFromContext =>
-  val assetsResolver = new AssetsResolver("/assets/", "assets.map", environment)
+  val assetsResolver = new AssetsResolver("", "assets.map", environment)
```
Comment (rupertbates) [inline app/wiring/Assets.scala]: The newest version of webpack-manifest-plugin includes  `/assets/` in the output so we need to remove this base path or it gets doubled up

**Comment by svillafe** [comment]
@rupertbates Can you add a screenshot before/after of the yarn build-prod command, to check if the upgrade had some impact on the final bundle size?

**Comment by rupertbates** [comment]
@svillafe See attached screenshot

**Comment by svillafe** [comment]
@rupertbates Great Rupert, it looks great!! 👍

**Pull request merged**